homeassistant/components/vivotek/camera.py

```diff
@@ -0,0 +1,123 @@
+"""Support for Vivotek IP Cameras."""
+
+import logging
+
+import voluptuous as vol
+from libpyvivotek import VivotekCamera
+
+from homeassistant.const import (
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FRAMERATE = "framerate"
+
+DEFAULT_NAME = "Vivotek Camera"
+DEFAULT_EVENT_0_KEY = "event_i0_enable"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+        vol.Optional(CONF_FRAMERATE, default=2): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a Vivotek IP Camera."""
+    args = dict(
+        config=config,
+        cam=VivotekCamera(
+            host=config[CONF_IP_ADDRESS],
+            port=(443 if config[CONF_SSL] else 80),
+            verify_ssl=config[CONF_VERIFY_SSL],
+            usr=config[CONF_USERNAME],
+            pwd=config[CONF_PASSWORD],
+        ),
+        stream_source=(
+            "rtsp://%s:%s@%s:554/live.sdp",
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            config[CONF_IP_ADDRESS],
+        ),
+    )
+    add_entities([VivotekCam(**args)])
+
+
+class VivotekCam(Camera):
+    """A Vivotek IP camera."""
+
+    def __init__(self, config, cam, stream_source):
+        """Initialize a Vivotek camera."""
+        super().__init__()
+
+        self._frame_interval = 1 / config[CONF_FRAMERATE]
+        self._name = config[CONF_NAME]
+        self._motion_detection_enabled = False
+
+        self._stream_source = stream_source
+
+        self._brand = "Vivotek"
+        self._supported_features = SUPPORT_STREAM
```

Comment:
We don't need instance attributes for brand and supported features. Just return those constants directly in the respective properties.